.github/workflows/sonar.yml

```diff
@@ -0,0 +1,20 @@
+name: Build
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, reopened]
```

Comment:
Do we need this on `reopened` as well? When one is reopening a PR (if ever) I would assume there's not any new commit to analyze.

Reply:
Copied from the suggested code in the docs. You are right.